### ISSUE TITLE
Handling unspecified interested parties (hanging edges) in BODS data

### ISSUE THREAD

**Issue by kd-ods** [openownership/visualisation-tool#157]
In BODS 0.4, relationships (edges) are specified as existing between a `subject` (usually represented by a recordId (string value)) and an `interestedParty` (usually represented by a recordId(string value)).

However, sometimes an interested party (or a subject) might not be provided in the data, in which case the value for the `interestedParty` is an [unspecified record object](https://standard.openownership.org/en/latest/standard/reference.html#unspecifiedrecord).

This is the case in [this example file](https://github.com/openownership/data-standard/blob/main/examples/listed-company-exempt-from-disclosure.json):

```
[...]
    "recordStatus": "new",
    "recordType": "relationship",
    "recordDetails": {
      "isComponent": false,
      "subject": "4c7ea3bfbe6c",
      "interestedParty": {
        "reason": "subjectExemptFromDisclosure",
        "description": "Exempt from disclosure as a company listed on a recognised stock exchange"
      }
    }
[...]
```
An error like this is currently generated when submitting that example file:

```
Uncaught runtime errors:
ERROR
n is undefined
./node_modules/dagre-d3/lib/render.js/e.exports/m/<@http://localhost:9000/main.js:2:574459
./node_modules/dagre-d3/lib/render.js/e.exports/m/<@http://localhost:9000/main.js:2:574402
m@http://localhost:9000/main.js:2:575087
./src/index.js/c<@http://localhost:9000/main.js:2:57176
p@http://localhost:9000/main.js:2:53749
./demo/index.js/e/<@http://localhost:9000/main.js:2:53572
c@http://localhost:9000/main.js:2:773535
./node_modules/regenerator-runtime/runtime.js/u/i._invoke@http://localhost:9000/main.js:2:773289
./node_modules/regenerator-runtime/runtime.js/_/</<@http://localhost:9000/main.js:2:773894
t@http://localhost:9000/main.js:2:524
a@http://localhost:9000/main.js:2:736
```

#### Suggested handling

If a relationship statement's `interestedParty` or `subject` value is an object (not a string):

- create a special 'Unspecified' node which renders like so (question mark in a stack of dotted circles), plus a label of 'Unspecified':

![image](https://github.com/user-attachments/assets/0f94c765-035d-4ff8-afa6-0971474558f0)
- render the edge as normal.

**Comment by kd-ods** [comment]
FWIW, the icon is from [BOVS](https://www.openownership.org/en/publications/beneficial-ownership-visualisation-system/unknowns/).

@codemacabre - how does the above suggested handling sound?

**Comment by codemacabre** [comment]
@kd-ods I think this is a good way to handle these hanging edges and shouldn't require particularly significant changes to the codebase to implement.

**Comment by kd-ods** [comment]
Also, there is already some related handling of unspecified interested parties in BODS 0.3 data by the visualiser. See #150. Worth noting that in the test BODS 0.3 data there, the `interestedParty` property is given as:

```
"interestedParty": {"reason":"interestedPartyExemptFromDisclosure"}
```

Which is actually invalid BODS 0.3! To be valid it should be:

```
"interestedParty": {
    "unspecified": {"reason":"interestedPartyExemptFromDisclosure"}
}
```
Correcting the data in #150 does not change the rendering or throw an error, suggesting that the current code is flexible enough to handle a common error in the data.

To be clear: we have now closed #150 in favour of this issue. And the scope of this issue is to handle unspecified records in BODS 0.4 data. I don't think we should change or correct the rendering of unspecified info in pre BODS 0.4 data.

**Comment by codemacabre** [comment]
This is how the example file renders now following the changes in #162:
![bods](https://github.com/user-attachments/assets/198ebae2-4eb6-42f7-b9e9-6feeb337eb20)
